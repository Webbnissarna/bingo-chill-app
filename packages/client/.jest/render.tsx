import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
