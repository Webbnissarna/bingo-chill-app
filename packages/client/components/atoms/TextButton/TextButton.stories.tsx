import type { Story, StoryDefault } from "@ladle/react";
import TextButton from ".";

interface StoryProps {
  text: string;
}

function StoryComponent({ text }: StoryProps) {
  return <TextButton text={text} onClick={() => alert("Clicked!")} />;
}

export const Default: Story<StoryProps> = StoryComponent;

Default.args = {
  text: "Hello World",
};

Default.argTypes = {
  text: {
    control: { type: "text" },
  },
};

export default {
  title: "Atoms / Text Button",
} satisfies StoryDefault;
