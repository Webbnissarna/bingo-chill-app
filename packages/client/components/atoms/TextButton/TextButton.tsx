interface TextButtonProps {
  text: string;
  onClick: () => void;
}

export default function TextButton({
  text,
  onClick,
}: TextButtonProps): JSX.Element {
  return (
    <button
      className="font-text font-base text-snowStorm-2 bg-polarNight-3 px-3 py-1 rounded-lg hover:bg-polarNight-2 active:bg-polarNight-1 transition"
      onClick={onClick}
    >
      {text}
    </button>
  );
}
