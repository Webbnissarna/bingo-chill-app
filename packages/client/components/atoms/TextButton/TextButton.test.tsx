import { render } from "../../../.jest/render";
import TextButton from ".";

describe("TextButton", () => {
  it("renders", () => {
    const { asFragment } = render(
      <TextButton text="Hello World" onClick={() => undefined} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
