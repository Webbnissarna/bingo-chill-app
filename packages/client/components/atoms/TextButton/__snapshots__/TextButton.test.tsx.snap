// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton renders 1`] = `
<DocumentFragment>
  <button
    class="font-text font-base text-snowStorm-2 bg-polarNight-3 px-3 py-1 rounded-lg hover:bg-polarNight-2 active:bg-polarNight-1 transition"
  >
    Hello World
  </button>
</DocumentFragment>
`;
